src/librbd/io/ImageDispatcherInterface.h

```diff
@@ -29,6 +29,7 @@ struct ImageDispatcherInterface
   virtual void wait_on_writes_unblocked(Context *on_unblocked) = 0;
 
   virtual void invalidate_cache(Context* on_finish) = 0;
+  virtual void remap_extents(Extents&& image_extents, bool reverse) = 0;
```

Comment:
Nit: perhaps use an enum instead of the bool just so the code is clearer as to the desired behavior when invoking `remap_extents`.